common/types/mapper/thrift/shared.go

```diff
@@ -1556,6 +1567,17 @@ func ToDescribeHistoryHostRequest(t *shared.DescribeHistoryHostRequest) *types.D
 	}
 }
 
+// ToDescribeShardDistributionRequest converts thrift DescribeHistoryHostRequest type to internal
+func ToDescribeShardDistributionRequest(t *shared.DescribeShardDistributionRequest) *types.DescribeShardDistributionRequest {
+	if t == nil {
+		return nil
+	}
+	return &types.DescribeShardDistributionRequest{
+		PageSize: *t.PageSize,
```

Comment:
Nit: use t.GetPageSize()/t.GetPageID() instead to avoid nil pointer dereference.